simple-js-examples/simple-fota-example.js

```diff
@@ -0,0 +1,545 @@
+			{ deviceId: deviceId },
+			function (error, response) {
+				clearTimeout(timeout);
+				if (!error) {
+					resolve(response);
+				} else {
+					reject(error);
+				}
+			}
+		);
+	});
+};
+
+EdgePTExample.prototype.updateExampleDeviceResources = async function (
+	deviceId
+) {
+	let self = this;
+	return new Promise((resolve, reject) => {
+		params = self._createDeviceParams(
+			deviceId,
+			19.5 /* temp */,
+			20.5 /* set point */,
+			"0.0.0",
+		);
+
+		let timeout = setTimeout(() => {
+			reject("Timeout");
+		}, TIMEOUT);
+
+		self.client.send("write", params, function (error, response) {
+			clearTimeout(timeout);
+			if (!error) {
+				resolve(response);
+			} else {
+				reject(error);
+			}
+		});
+	});
+};
+
+EdgePTExample.prototype.updateExampleManifestResources = async function (
+	deviceId,
+	fw_version
+	) {
+	let self = this;
+	return new Promise((resolve, reject) => {
+		params = self._createDeviceParams(deviceId,19.5,20.5,fw_version)
+		let timeout = setTimeout(() => {
+		reject("Timeout");
+		}, TIMEOUT);
+	
+		self.client.send("write", params, function (error, response) {
+		clearTimeout(timeout);
+		if (!error) {
+			resolve(response);
+		} else {
+			reject(error);
+		}
+		});
+	});
+	};
+
+EdgePTExample.prototype.exposeVendorandClass = function () {
+	let self = this;
+	self.client.expose("manifest_meta_data", (params, response) => {
+		// checking the request
+		if (params.uri.deviceId == null || params.classid == null || params.vendorid ==null || params.component_name == null) {
+			console.log(RED,"Missing any of the filed in params: classid, deviceId, vendorid , url, version, component_name");
+			response({
+			"code": -32602,
+			"data": "Missing any of the filed in params: classid, deviceId, vendorid, version, component name",
+			"message": "Missing any of the filed in params: classid, deviceId, vendorid, version, component name"
+			},  null);
+			return;
+		}
+		console.log(GREEN, "received fota request")
+		console.log(params)
+		var fw_size = params.size
+		var classid = new Buffer.from(params.classid, "base64");
+		var vendorid = new Buffer.from(params.vendorid, "base64");
+		var b_64version = params.version
+		var data = new Buffer.from(b_64version, "base64");
+		var fw_version = data.toString("ascii")
+		var deviceid = params.uri.deviceId;
+		var send_data = {
+			deviceId: deviceid,
+			size: fw_size
+		}
+		// checking vendor and class id from manifest is equal or not.
+		if((Buffer.compare(vendorid, VENDORID)!= 0) && (Buffer.compare(classid, CLASSID)!= 0)) {
+			response({
+				"code": -32602,
+				"data": "wrong vendor or class ID",
+				"message": "wrong vendor or class ID"
+			}, /* success */ null);
+		}
+		else {
+			response( null,'ok');
+			let timeout = setTimeout(() => {
+							reject("Timeout");
+							}, TIMEOUT_FOR_FOTA);
+
+			self.client.send("download_asset", send_data ,function (error, response) {
+				clearTimeout(timeout);
+				if (!error) {
+					//console.log(response)
```

Comment:
```suggestion
```

Reply:
corrected this in the new commit.  Thanks for the comment.